### PR TITLE
Implement server side validations for files only admins can change. I…

### PR DESCRIPTION
**Pull request by garethbowen**
…ssue: #2868

### CODE REVIEW

File: tests/nodeunit/unit/validate_doc_update.js [hunk 1]
```diff
@@ -29,22 +19,64 @@ exports.setUp = function(cb) {
...
+exports['only db and national admins are allowed change forms'] = function(test) {
+  var doc = { type: 'form' };
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change translations'] = function(test) {
+  var doc = { type: 'form' };
```
Comment (SCdF) [inline tests/nodeunit/unit/validate_doc_update.js]: erm

File: tests/nodeunit/unit/validate_doc_update.js [hunk 2]
```diff
@@ -29,22 +19,64 @@ exports.setUp = function(cb) {
...
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change resources'] = function(test) {
+  var doc = { type: 'form' };
```
Comment (SCdF) [inline tests/nodeunit/unit/validate_doc_update.js]: uhhhh

File: tests/nodeunit/unit/validate_doc_update.js [hunk 3]
```diff
@@ -29,22 +19,64 @@ exports.setUp = function(cb) {
...
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change appcache doc'] = function(test) {
+  var doc = { type: 'form' };
```
Comment (SCdF) [inline tests/nodeunit/unit/validate_doc_update.js]: ahem.

Sir. Once one copies and once one pastes, one must also edit for relevance!

Comment (garethbowen) [inline tests/nodeunit/unit/validate_doc_update.js]: But the tests were green!  :(

Fixed.

**Pull request merged**